panic in `no-misused-promises` on empty JSX function attributes

## Versions

* `oxlint-tsgolint` - 0.22.1
* `oxlint` - 1.62.0

## Reproduction code

```tsx
function MyComponent() {
    return <div onClick={} />
}
```

It seems that panic only occurs for callback attributes: for example, `className={}` instead of `onClick={}` does not cause panic.

## Stacktrace

<details>

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x7ff7525933f2]

goroutine 6239 [running]:
sync.(*WaitGroup).Go.func1.1()
        /opt/hostedtoolcache/go/1.26.2/x64/src/sync/waitgroup.go:251 +0x45
panic({0x7ff75282d340?, 0x7ff7535c3440?})
        /opt/hostedtoolcache/go/1.26.2/x64/src/runtime/panic.go:860 +0x13a
github.com/microsoft/typescript-go/internal/ast.IsSourceFile(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast_generated.go:8197
github.com/microsoft/typescript-go/internal/checker.(*Checker).getTypeOfNode(0x9ae1f044580?, 0x9ae00000000?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:31132 +0x12
github.com/microsoft/typescript-go/internal/checker.(*Checker).GetTypeAtLocation(0x9ae2ce9d308, 0x1?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:31359 +0x28
github.com/typescript-eslint/tsgolint/internal/rules/no_misused_promises.init.func1.3(0x0)
        /home/runner/work/tsgolint/tsgolint/internal/rules/no_misused_promises/no_misused_promises.go:130 +0x34
github.com/typescript-eslint/tsgolint/internal/rules/no_misused_promises.init.func1.11(0x9ae1e7b3e00)
        /home/runner/work/tsgolint/tsgolint/internal/rules/no_misused_promises/no_misused_promises.go:326 +0xc2
github.com/typescript-eslint/tsgolint/internal/rules/no_misused_promises.init.func1.30(0x7ff75283ae20?)
        /home/runner/work/tsgolint/tsgolint/internal/rules/no_misused_promises/no_misused_promises.go:768 +0x2b
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.1(0x124, 0x9ae1e7b3e00)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:379 +0x5b
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.3(0x9ae1e7b3e00)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:426 +0x42
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:38
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:47
github.com/microsoft/typescript-go/internal/ast.(*JsxAttributes).ForEachChild(0x9ae2cd7c480?, 0x9ae2f7480c0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast_generated.go:6587 +0x67
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x125?, 0x9ae1e7b3e50?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:194 +0x1c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.3(0x9ae1e7b3e50)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:441 +0x125
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:31
github.com/microsoft/typescript-go/internal/ast.(*JsxSelfClosingElement).ForEachChild(0x9ae1e7b3ea0, 0x9ae2f7480c0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast_generated.go:6718 +0x91
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x11e?, 0x9ae1e7b3ea0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:194 +0x1c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.3(0x9ae1e7b3ea0)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:441 +0x125
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:31
github.com/microsoft/typescript-go/internal/ast.(*ReturnStatement).ForEachChild(0x9ae2cd7c480?, 0x9ae2dda3638?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast_generated.go:1327 +0x23
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x9ae1fe900fe?, 0x9ae1ea1b5f0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:194 +0x1c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.3(0x9ae1ea1b5f0)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:441 +0x125
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:38
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:47
github.com/microsoft/typescript-go/internal/ast.(*Block).ForEachChild(0x9ae2cd7c480?, 0x9ae2f7480c0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast_generated.go:1798 +0x67
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x9ae1fe900f2?, 0x9ae1ec54240?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:194 +0x1c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.3(0x9ae1ec54240)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:441 +0x125
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:31
github.com/microsoft/typescript-go/internal/ast.(*FunctionDeclaration).ForEachChild(0x9ae1f708600, 0x9ae2f7480c0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast_generated.go:2192 +0x191
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x107?, 0x9ae1f708608?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:194 +0x1c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.3(0x9ae1f708608)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:441 +0x125
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:38
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:47
github.com/microsoft/typescript-go/internal/ast.(*SourceFile).ForEachChild(0x9ae1e94b188, 0x9ae2f7480c0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:2595 +0x9b
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x7ff75283ae20?, 0x9ae2dd980c0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:194 +0x1c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2()
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:449 +0x90e
github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue.func1()
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/core/workgroup.go:40 +0x13
sync.(*WaitGroup).Go.func1()
        /opt/hostedtoolcache/go/1.26.2/x64/src/sync/waitgroup.go:258 +0x4a
created by sync.(*WaitGroup).Go in goroutine 1
        /opt/hostedtoolcache/go/1.26.2/x64/src/sync/waitgroup.go:238 +0x73
Error running tsgolint: "exit status: exit code: 2"
```

</details>

## Config

<details>

```json
{
  "plugins": [
    "react",
    "typescript",
    "import",
    "jsx-a11y"
  ],
  "categories": {
    "correctness": "deny"
  },
  "rules": {
    "constructor-super": "deny",
    "curly": "warn",
    "for-direction": "deny",
    "getter-return": "deny",
    "no-array-constructor": "deny",
    "no-async-promise-executor": "deny",
    "no-caller": "deny",
    "no-case-declarations": "deny",
    "no-class-assign": "deny",
    "no-compare-neg-zero": "deny",
    "no-cond-assign": "deny",
    "no-const-assign": "deny",
    "no-constant-binary-expression": "deny",
    "no-constant-condition": "deny",
    "no-control-regex": "deny",
    "no-debugger": "deny",
    "no-delete-var": "deny",
    "no-dupe-class-members": "deny",
    "no-dupe-else-if": "deny",
    "no-dupe-keys": "deny",
    "no-duplicate-case": "deny",
    "no-empty": "deny",
    "no-empty-character-class": "deny",
    "no-empty-pattern": "deny",
    "no-empty-static-block": "deny",
    "no-eval": "deny",
    "no-ex-assign": "deny",
    "no-extra-boolean-cast": "deny",
    "no-fallthrough": "deny",
    "no-func-assign": "deny",
    "no-global-assign": "deny",
    "no-import-assign": "deny",
    "no-invalid-regexp": "deny",
    "no-irregular-whitespace": "deny",
    "no-iterator": "deny",
    "no-loss-of-precision": "deny",
    "no-misleading-character-class": "deny",
    "no-new-native-nonconstructor": "deny",
    "no-nonoctal-decimal-escape": "deny",
    "no-obj-calls": "deny",
    "no-prototype-builtins": "deny",
    "no-redeclare": "deny",
    "no-regex-spaces": "deny",
    "no-self-assign": "deny",
    "no-setter-return": "deny",
    "no-shadow-restricted-names": "deny",
    "no-sparse-arrays": "deny",
    "no-this-before-super": "deny",
    "no-unassigned-vars": "deny",
    "no-unexpected-multiline": "deny",
    "no-unreachable": "deny",
    "no-unsafe-finally": "deny",
    "no-unsafe-negation": "deny",
    "no-unsafe-optional-chaining": "deny",
    "no-unused-expressions": [
      "deny",
      [
        {
          "enforceForJSX": true
        }
      ]
    ],
    "no-unused-labels": "deny",
    "no-unused-private-class-members": "deny",
    "no-unused-vars": [
      "deny",
      [
        {
          "argsIgnorePattern": "^_",
          "ignoreRestSiblings": true
        }
      ]
    ],
    "no-useless-backreference": "deny",
    "no-useless-catch": "deny",
    "no-useless-escape": "deny",
    "no-useless-rename": "deny",
    "no-with": "deny",
    "require-yield": "deny",
    "use-isnan": "deny",
    "valid-typeof": "deny",
    "import/default": "deny",
    "import/namespace": "deny",
    "import/no-cycle": "deny",
    "jsx_a11y/alt-text": "deny",
    "jsx_a11y/anchor-has-content": "deny",
    "jsx_a11y/anchor-is-valid": "deny",
    "jsx_a11y/aria-activedescendant-has-tabindex": "deny",
    "jsx_a11y/aria-props": "warn",
    "jsx_a11y/aria-proptypes": "deny",
    "jsx_a11y/aria-role": "deny",
    "jsx_a11y/aria-unsupported-elements": "warn",
    "jsx_a11y/autocomplete-valid": "deny",
    "jsx_a11y/click-events-have-key-events": "deny",
    "jsx_a11y/heading-has-content": "deny",
    "jsx_a11y/html-has-lang": "deny",
    "jsx_a11y/iframe-has-title": "deny",
    "jsx_a11y/img-redundant-alt": "deny",
    "jsx_a11y/label-has-associated-control": "deny",
    "jsx_a11y/lang": "deny",
    "jsx_a11y/media-has-caption": "deny",
    "jsx_a11y/mouse-events-have-key-events": "deny",
    "jsx_a11y/no-access-key": "deny",
    "jsx_a11y/no-aria-hidden-on-focusable": "deny",
    "jsx_a11y/no-autofocus": "deny",
    "jsx_a11y/no-distracting-elements": "deny",
    "jsx_a11y/no-noninteractive-tabindex": [
      "deny",
      [
        {
          "tags": [],
          "roles": [
            "tabpanel"
          ],
          "allowExpressionValues": true
        }
      ]
    ],
    "jsx_a11y/no-redundant-roles": "deny",
    "jsx_a11y/no-static-element-interactions": "deny",
    "jsx_a11y/prefer-tag-over-role": "deny",
    "jsx_a11y/role-has-required-aria-props": "warn",
    "jsx_a11y/role-supports-aria-props": "warn",
    "jsx_a11y/scope": "deny",
    "jsx_a11y/tabindex-no-positive": "deny",
    "react/exhaustive-deps": "warn",
    "react/forward-ref-uses-ref": "deny",
    "react/jsx-key": "deny",
    "react/jsx-no-comment-textnodes": "warn",
    "react/jsx-no-duplicate-props": "deny",
    "react/jsx-no-script-url": "warn",
    "react/jsx-no-undef": "deny",
    "react/jsx-props-no-spread-multi": "deny",
    "react/no-array-index-key": "warn",
    "react/no-children-prop": "allow",
    "react/no-danger": "warn",
    "react/no-danger-with-children": "deny",
    "react/no-did-mount-set-state": "deny",
    "react/no-did-update-set-state": "deny",
    "react/no-direct-mutation-state": "deny",
    "react/no-find-dom-node": "deny",
    "react/no-is-mounted": "deny",
    "react/no-redundant-should-component-update": "deny",
    "react/no-render-return-value": "deny",
    "react/no-string-refs": "deny",
    "react/no-this-in-sfc": "deny",
    "react/no-unsafe": "warn",
    "react/no-will-update-set-state": "deny",
    "react/rules-of-hooks": "deny",
    "react/self-closing-comp": "deny",
    "react/void-dom-elements-no-children": "deny",
    "typescript/await-thenable": "deny",
    "typescript/ban-ts-comment": "warn",
    "typescript/consistent-type-imports": [
      "deny",
      [
        {
          "prefer": "type-imports",
          "fixStyle": "inline-type-imports"
        }
      ]
    ],
    "typescript/no-array-delete": "deny",
    "typescript/no-base-to-string": "deny",
    "typescript/no-duplicate-enum-values": "deny",
    "typescript/no-duplicate-type-constituents": "deny",
    "typescript/no-empty-object-type": [
      "deny",
      [
        {
          "allowInterfaces": "with-single-extends"
        }
      ]
    ],
    "typescript/no-explicit-any": "warn",
    "typescript/no-extra-non-null-assertion": "deny",
    "typescript/no-floating-promises": "deny",
    "typescript/no-for-in-array": "deny",
    "typescript/no-implied-eval": "deny",
    "typescript/no-meaningless-void-operator": "deny",
    "typescript/no-misused-new": "deny",
    "typescript/no-misused-promises": "deny",
    "typescript/no-misused-spread": "deny",
    "typescript/no-namespace": [
      "deny",
      [
        {
          "allowDeclarations": true
        },
        "error",
        {
          "allowDeclarations": true,
          "allowDefinitionFiles": true
        }
      ]
    ],
    "typescript/no-non-null-asserted-optional-chain": "deny",
    "typescript/no-redundant-type-constituents": "deny",
    "typescript/no-require-imports": "deny",
    "typescript/no-this-alias": "deny",
    "typescript/no-unnecessary-parameter-property-assignment": "deny",
    "typescript/no-unnecessary-type-assertion": "deny",
    "typescript/no-unnecessary-type-constraint": "deny",
    "typescript/no-unsafe-argument": "deny",
    "typescript/no-unsafe-assignment": "deny",
    "typescript/no-unsafe-call": "deny",
    "typescript/no-unsafe-declaration-merging": "deny",
    "typescript/no-unsafe-enum-comparison": "deny",
    "typescript/no-unsafe-function-type": "deny",
    "typescript/no-unsafe-member-access": "deny",
    "typescript/no-unsafe-return": "deny",
    "typescript/no-unsafe-unary-minus": "deny",
    "typescript/no-useless-default-assignment": "deny",
    "typescript/no-useless-empty-export": "deny",
    "typescript/no-wrapper-object-types": "deny",
    "typescript/only-throw-error": [
      "deny",
      [
        {
          "allow": [
            {
              "from": "package",
              "package": "@tanstack/router-core",
              "name": [
                "Redirect",
                "NotFoundError",
                "redirect",
                "notFound"
              ]
            },
            {
              "from": "package",
              "package": "@tanstack/react-router",
              "name": [
                "Redirect",
                "NotFoundError",
                "redirect",
                "notFound"
              ]
            }
          ]
        }
      ]
    ],
    "typescript/prefer-as-const": "deny",
    "typescript/prefer-namespace-keyword": "deny",
    "typescript/prefer-promise-reject-errors": "deny",
    "typescript/require-array-sort-compare": "deny",
    "typescript/require-await": "deny",
    "typescript/restrict-plus-operands": "deny",
    "typescript/restrict-template-expressions": "deny",
    "typescript/triple-slash-reference": "deny",
    "typescript/unbound-method": "deny"
  },
  "settings": {
    "jsx-a11y": {
      "polymorphicPropName": null,
      "components": {},
      "attributes": {}
    },
    "next": {
      "rootDir": []
    },
    "react": {
      "formComponents": [],
      "linkComponents": [],
      "version": null,
      "componentWrapperFunctions": []
    },
    "jsdoc": {
      "ignorePrivate": false,
      "ignoreInternal": false,
      "ignoreReplacesDocs": true,
      "overrideReplacesDocs": true,
      "augmentsExtendsReplacesDocs": false,
      "implementsReplacesDocs": false,
      "exemptDestructuredRootsFromChecks": false,
      "tagNamePreference": {}
    },
    "vitest": {
      "typecheck": false
    },
    "jest": {
      "version": null
    }
  },
  "env": {
    "browser": true,
    "builtin": true,
    "es2026": true
  },
  "globals": {},
  "overrides": [
    {
      "files": [
        "**/*.ts",
        "**/*.tsx",
        "**/*.mts",
        "**/*.cts"
      ],
      "env": null,
      "globals": null,
      "plugins": null,
      "rules": {
        "no-class-assign": "allow",
        "no-const-assign": "allow",
        "no-dupe-class-members": "allow",
        "no-dupe-keys": "allow",
        "no-func-assign": "allow",
        "no-import-assign": "allow",
        "no-new-native-nonconstructor": "allow",
        "no-obj-calls": "allow",
        "no-redeclare": "allow",
        "no-setter-return": "allow",
        "no-this-before-super": "allow",
        "no-unsafe-negation": "allow",
        "no-var": "deny",
        "no-with": "allow",
        "prefer-rest-params": "deny",
        "prefer-spread": "deny"
      }
    }
  ],
  "options": {
    "typeAware": true
  },
  "ignorePatterns": [
    "src/app/router/routeTree.gen.ts",
    "src/app/typings/env.d.ts",
    "src/shared/api/code-gen"
  ]
}
```

</details>

--- a/internal/rules/no_misused_promises/no_misused_promises.go
+++ b/internal/rules/no_misused_promises/no_misused_promises.go
@@ -322,6 +322,9 @@ var NoMisusedPromisesRule = rule.Rule{
 			}
 			expressionContainer := node.Initializer.AsJsxExpression()
 			expression := expressionContainer.Expression
+			if expression == nil {
+				return
+			}
 			contextualType := checker.Checker_getContextualType(ctx.TypeChecker, node.Initializer, checker.ContextFlagsNone)
 			if contextualType != nil && isVoidReturningFunctionType(node.Initializer, contextualType) && returnsThenable(expression) {
 				ctx.ReportNode(node.Initializer, buildVoidReturnAttributeMessage())

--- a/internal/rules/no_misused_promises/no_misused_promises_test.go
+++ b/internal/rules/no_misused_promises/no_misused_promises_test.go
@@ -258,6 +258,15 @@ const Component: any = () => null;
 		},
 		{
 			Code: `
+declare namespace JSX {
+  interface IntrinsicElements { div: { onClick?: () => void }; }
+}
+function MyComponent() { return <div onClick={} />; }
+      `,
+			Tsx: true,
+		},
+		{
+			Code: `
 interface ItLike {
   (name: string, callback: () => Promise<void>): void;
   (name: string, callback: () => void): void;

